src/mountscript.sh

```diff
@@ -217,13 +225,48 @@ get_dir_from_share()
 
     if [ -z "$account" -o -z "$container" -o -n "$extra" ]; then
         eecho "Bad share name: ${hostshare}."
-        eecho "Share to be mounted must be of the form 'account.blob.core.windows.net:/account/container'."
+        eecho "Share to be mounted must be of the form 'account.$2.core.windows.net:/account/container'."
         return 1
     fi
 
     echo "$share"
 }
 
+get_version_from_MOUNT_OPTIONS()
+{
+    local mount_options="$MOUNT_OPTIONS"
+    local ver_string="vers="
+    local minor_ver_string="minorversion"
+    local nfs_ver=""
+    local nfs_minorvers=""
+
+    if [ -z "$mount_options" ] || [[ ! "$mount_options" == *"$ver_string"* ]]; then
+        eecho "Bad mount options: ${mount_options}."
+        eecho "Share to be mounted must have version '-o vers=4.1'."
```

Comment:
>eecho "Share to be mounted must have version '-o vers=4.1'." [](http://example.com/codeflow?start=8&length=60)

can we not default version to 4.1 if tls/notls is specified?

Reply:
chatted with Richa offline, the right thing to do is to default to v3 if vers is not specified.